### ISSUE TITLE
Proposal 2: implement LockedValue RWLockedValue which provides interface like rust mutex

### ISSUE THREAD

**Issue by dashjay** [dashjay/gog#2]
I like [rust mutex interface](https://doc.rust-lang.org/std/sync/struct.Mutex.html) so much, usually we can only guess the relationship of locker and data. The rust mutex implementation is very clear, we can easily know our data protect by which lock.

I'm not trying to draw a comparison between rust and golang, I just think the implementation is very clear and worth learning from.
```rust
let data = Arc::new(Mutex::new(0));
{
    let mut data = data.lock().unwrap();
...
}
```
Please comment if you think there are better design on function name or implement.

## Interface

```golang
type LockedValue interface{
    Lock() T
    Unlock()
    TryLock() T, bool
    SetValue(in T)
}


type RWLockedValue interface{
    Lock() T
    UnLock()
    TryLock() T, bool
    RLock() T
    RUnlock()
    RTryLock() T, bool
    
    SetValue(in T)
}

```

**Comment by dashjay** [comment]
#6

**Comment by dashjay** [comment]
Resolved by #6
